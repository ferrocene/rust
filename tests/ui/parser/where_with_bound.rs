//@revisions: edition2015 postedition2015
//@[edition2015] edition: 2015
//@[postedition2015] edition: 2018..

fn foo<T>() where <T>::Item: ToString, T: Iterator { }
//~^ ERROR generic parameters on `where` clauses are reserved for future use
//[edition2015]~| ERROR cannot find type `Item` in the crate root
//[postedition2015]~| ERROR cannot find crate `Item` in the list of imported crates

fn main() {}
