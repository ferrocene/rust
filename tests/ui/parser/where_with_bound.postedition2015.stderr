error: generic parameters on `where` clauses are reserved for future use
  --> $DIR/where_with_bound.rs:5:19
   |
LL | fn foo<T>() where <T>::Item: ToString, T: Iterator { }
   |                   ^^^ currently unsupported

error[E0412]: cannot find crate `Item` in the list of imported crates
  --> $DIR/where_with_bound.rs:5:24
   |
LL | fn foo<T>() where <T>::Item: ToString, T: Iterator { }
   |                        ^^^^ not found in the list of imported crates

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0412`.
